Standalone/PythonTests/Automated/benchmark_data_aggregator.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python
+"""
+Copyright (c) Contributors to the Open 3D Engine Project. For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+from argparse import ArgumentParser
+import json
+from pathlib import Path
+import time
+import copy
+import subprocess
+import requests
+
+from ly_test_tools.mars.filebeat_client import FilebeatClient
+
+class BenchmarkDataAggregator:
+    def __init__(self, workspace, logger, test_suite):
+        self.build_dir = Path(workspace.paths.build_directory())
+        self.results_dir = Path(workspace.paths.project()) / 'user' / 'Scripts' / 'PerformanceBenchmarks'
+        self.test_suite = test_suite
+        self.logger = logger
+        self.filebeat_client = FilebeatClient(logger)
+
+    def _process_benchmark(self, benchmark_dir):
+        benchmark_metadata = {}
+        frame_stats = { 'count': 0, 'totalTime': 0 }
+        pass_stats = {}
+
+        metadata_file = benchmark_dir / 'benchmark_metadata.json'
+        if metadata_file.exists():
+            data = json.loads(metadata_file.read_text())
+            benchmark_metadata = data['ClassData']
+        else:
+            self.logger.warn(f'Metadata file could not be found at {metadata_file}')
+
+        # this allows us to add additional data if necessary, e.g. frame_test_timestamps.json
+        is_timestamp_file = lambda file: file.name.startswith('frame') and file.name.endswith('_timestamps.json')
+
+        for file in benchmark_dir.iterdir():
+            if file.is_dir() or not is_timestamp_file(file):
+                continue
+
+            data = json.loads(file.read_text())
+            entries = data['ClassData']['timestampEntries']
+
+            frame_time = 0
+            for entry in entries:
```

Comment:
since this is a `for` loop inside of a `for` loop, its probably better we split into its own function that returns `time_ns` for updating `frame_time`
in general we try to avoid nested `for` loops and split them into new functions with returns instead (makes it easier to follow the code)

also, if we could declare the variables related to that code closer to the code, its easier to follow
i.e. `pass_stats ` is on line 29 and i had to find it to see what this `pass_entry = pass_stats.get(name, { 'totalTime': 0, 'maxTime': 0 })` call was doing

speaking of which, isn't `pass_stats` always `{}` before this call is made? that makes the `.get()` call redundant since it won't get anything matching `entry['passName']` inside `pass_stats = {}`

you should just do this and remove the `.get()` call if `pass_stats` is always `{}` before that call is made:
`pass_stats = {'totalTime': 0, 'maxTime': 0}`

Reply:
since its iterating and looping over values it seems the `pass_entry = pass_stats.get(name, { 'totalTime': 0, 'maxTime': 0 })` may be needed, but i would initialize `pass_entry = {'totalTime': 0, 'maxTime': 0}` before this for loop / future function instead of `pass_entry = {}`

Reply:
UPDATE: This is better but let's update it to not be nested (added more info in a new review comment)